From 062ad785d5be352c016097ff6238925cd1280b6d Mon Sep 17 00:00:00 2001
From: MyungJoo Ham <myungjoo.ham@samsung.com>
Date: Wed, 27 Oct 2021 23:04:49 +0900
Subject: [PATCH] tensor-filter-common: unnecessary usage of volatile

GLib-Once (g_once_init_enter/leave) have already taken
care of concurrency.
Its users do not need to worry about cache coherency or
concurrency issues for the initialization.

Actually, as #3440 is showing, the volatile keyword
is neglected by g_once_init_enter function, making
the usage of volatile meaningless anyway.

Remove volatile keyword and mitigate the first
compiler warning-error of #3440

Signed-off-by: MyungJoo Ham <myungjoo.ham@samsung.com>

The patch was imported from the upstream github server
  https://github.com/nnstreamer/nnstreamer
  main branch
  commit c263f96

Signed-off-by: Julien Vuillaumier <julien.vuillaumier@nxp.com>
---
 gst/nnstreamer/tensor_filter/tensor_filter_common.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/gst/nnstreamer/tensor_filter/tensor_filter_common.c b/gst/nnstreamer/tensor_filter/tensor_filter_common.c
index 74f5bbe4..9f08c447 100644
--- a/gst/nnstreamer/tensor_filter/tensor_filter_common.c
+++ b/gst/nnstreamer/tensor_filter/tensor_filter_common.c
@@ -2762,9 +2762,9 @@ parse_accl_hw_fill (parse_accl_args accl_args)
 static GType
 accl_hw_get_type (void)
 {
-  static volatile gsize g_accl_hw_type_id__volatile = 0;
+  static gsize g_accl_hw_type_id_store = 0;
 
-  if (g_once_init_enter (&g_accl_hw_type_id__volatile)) {
+  if (g_once_init_enter (&g_accl_hw_type_id_store)) {
     static const GEnumValue values[] = {
       {ACCL_NONE, ACCL_NONE_STR, ACCL_NONE_STR},
       {ACCL_DEFAULT, ACCL_DEFAULT_STR, ACCL_DEFAULT_STR},
@@ -2788,10 +2788,10 @@ accl_hw_get_type (void)
 
     GType g_accl_hw_type_id =
         g_enum_register_static (g_intern_static_string ("accl_hw"), values);
-    g_once_init_leave (&g_accl_hw_type_id__volatile, g_accl_hw_type_id);
+    g_once_init_leave (&g_accl_hw_type_id_store, g_accl_hw_type_id);
   }
 
-  return g_accl_hw_type_id__volatile;
+  return g_accl_hw_type_id_store;
 }
 
 /**
-- 
2.25.1

